rfcs/accordians-component.md

```diff
@@ -0,0 +1,127 @@
+# Accordion Component
+
+## Usage
+
+```js
+import {Accordion, AccordionItem, Root, Toggle} from './index';
+
+export default () => {
+  return (
+    <div>
+      <Accordion>
+        <AccordionItem key={1} title={'First title'}><div>First content</div></AccordionItem>
+        <AccordionItem key={2} title={'Second title'}><div>Second content</div></AccordionItem>
+      </Accordion>
+    </div>
+  );
+};
+````
+
+```js
+import {Accordion, AccordionItem, Root, Toggle} from './index';
+import {styled} from '../styles/index';
+
+const CustomRoot = styled(Root, props => ({
+  textColor: 'red',
+}));
+
+const CustomToggle = styled(Toggle, props => ({
+  backgroundColor: props.$isExpanded ? 'green' : 'blue',
+}));
+
+export default () => {
+  return (
+    <div>
+      <Accordion
+        onToggle={(event, index, isExpanded) => {console.log('Accordian ' + index + ' is ' + (isExpanded ? 'expanded' : 'collapsed') )}}
+        overrides={{
+          Root: CustomRoot,
+          Toggle: CustomToggle,
+        }}
+        expanded={[false, true]}
```

Comment:
For stateless Accordion, I'm wondering if switching to an `isOpen` prop on each `AccordionItem` is more straightforward. Internally AccordionItem could be implemented where the `isOpen` prop–if set–takes precedence over context from the accordion parent.

Reply:
And actually, I'm almost wondering if its even worth breaking up into `Accordion` and `StatefulAccordion`. Maybe `Accordion` should just be stateful by default, and the `isOpen` prop on AccordionItem can be used for control if needed. 

I know this is kind of a break from our existing pattern in other components, but I'm struggling to think of a case where I'd want the stateless accordion that isn't solved by the solution above. Having a single `Accordion` export could make things conceptually simpler.

Thoughts @blackswanny @nadiia @abmai ?

Reply:
only one should have the state. I would rather gather full state of accordion in one place than to separate it between smaller AccordionItems. Export is simpler, but the data is scattered among more instances instead of one.

Reply:
Cool, works for me.